custom_components/blueprint/sensor.py

```diff
@@ -1,28 +1,36 @@
-"""Sensor platform for blueprint."""
-from custom_components.blueprint.const import DEFAULT_NAME, DOMAIN, ICON, SENSOR
-from custom_components.blueprint.entity import BlueprintEntity
-
-
-async def async_setup_entry(hass, entry, async_add_devices):
-    """Setup sensor platform."""
-    coordinator = hass.data[DOMAIN][entry.entry_id]
-    async_add_devices([BlueprintSensor(coordinator, entry)])
-
-
-class BlueprintSensor(BlueprintEntity):
-    """blueprint Sensor class."""
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return f"{DEFAULT_NAME}_{SENSOR}"
-
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self.coordinator.data.get("static")
-
-    @property
-    def icon(self):
-        """Return the icon of the sensor."""
-        return ICON
+"""Sensor platform for Blueprint."""
+from custom_components.blueprint.const import DEFAULT_NAME, DOMAIN, ICON, SENSOR
+from custom_components.blueprint.entity import BlueprintEntity
+
+from homeassistant.util import slugify
+
+
+async def async_setup_entry(hass, entry, async_add_devices):
+    """Setup sensor platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_devices([BlueprintSensor(coordinator, entry)])
+
+
+class BlueprintSensor(BlueprintEntity):
+    """blueprint Sensor class."""
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{DEFAULT_NAME}_{SENSOR}"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        # slugify the state to allow translation
+        return slugify(self.coordinator.data.get("static"))
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return ICON
+
+    @property
+    def device_class(self):
+        """Return de device class of the sensor."""
+        return "blueprint__custom_device_class"
```

Comment:
This will not work, and when you have a device_class there should not be an icon

Reply:
Can you explain why it should not work ? Developer documentation ([here](https://developers.home-assistant.io/docs/internationalization/custom_integration) and[ here](https://developers.home-assistant.io/docs/internationalization/core#stringssensorjson)) is very light on the topic. 
For information this is working in my developer container.
And if it is a custom device class, where can I define the associated icon ?

Reply:
"blueprint__custom_device_class" is not a valid device_class.
For valid sensor device classes see <https://www.home-assistant.io/integrations/sensor/#device-class>

Reply:
I know this is not an official device class but the documentation is saying we have to create a **custom** device class. A least it is what I understand when reading [this section](https://developers.home-assistant.io/docs/internationalization/core#stringssensorjson) in the translation documentation. Do you understand it an other way ?

Reply:
TBH, state translations have never made sense to me, and it adds overhead to the maintainer.
at this point, I'm partial to just remove the sensor translations.

Reply:
As French native user, and when I see the size of the francophone community who is not confortable with English, I feel the state translation is useful in some specific integration.
You have more experience than me so your point of view is very valuable. I see more the blueprint as a complete template where user can delete useless feature for their context. But I agree state translation is uncommon.

If you really want to remove state translation, you should avoid using a sensor returning an English text as state.

Reply:
Even if we do not agree on the topic. You have rework the template by removing `sampleclient` and replace it by an API module in the custom component. The state of the sensor is now a random text from an API request so there is no reason to translate it. I will close this anr reopen an other for the basic strings file.